leaderelection/context.go

```diff
@@ -36,10 +36,13 @@ import (
 // WithDynamicLeaderElectorBuilder sets up the statefulset elector based on environment,
 // falling back on the standard elector.
 func WithDynamicLeaderElectorBuilder(ctx context.Context, kc kubernetes.Interface, cc ComponentConfig) context.Context {
-	ssc, err := NewStatefulSetConfig()
+	logger := logging.FromContext(ctx)
+	ssc, err := newStatefulSetConfig()
 	if err == nil {
-		return WithStatefulSetLeaderElectorBuilder(ctx, cc, *ssc)
+		logger.Info("Running with StatefulSet leader election")
```

Comment:
I think the logs here are redundant as in `buildElector` we have similar logs.

Reply:
🤷  it's once per start log, so should be ok.